test/unit_test_input_parser.c

```diff
@@ -1,6 +1,166 @@
 #include "unit_test_input_parser.h"
 
+// WARNING !!! This test may seg fault if it fails!
+int test_data(void) {
+  int i, j;
+  int size = 63;
+
+  GEN_LADDER_INPUT(size, size);
+  PARSE_FILE();
+
+  // in.data = (int **)malloc(sizeof(int*) * size);       // SANITY CHECK, THIS FORCES TEST TO PASS
+  for (i = 0; i < size; i++) {
+    // in.data[i] = (int *)malloc(sizeof(int) * (i + 1)); // SANITY CHECK, THIS FORCES TEST TO PASS
+    for (j = 0; j <= i; j++) {
+      // in.data[i][j] = j + 1;                           // SANITY CHECK, THIS FORCES TEST TO PASS
+      if ((ACT = in.data[i][j]) != (EXP = j + 1)) { 
+        FAIL("test_data")
+      }
+    }
+  }
+  PASS("test_data")
+}
+
+// WARNING !!! This test may seg fault if it fails!
```

Comment:
Does it have to segfault on failure?
Can't it check for those conditions, and just fail before segfaulting?